### ISSUE TITLE
Unconfigured auth type: facebook

### ISSUE THREAD

**Issue by jakobrosenberg** [rethinkdb/horizon#561]
Using Horizon v1.1.1 I get the following error: `Uncaught Error: Unconfigured auth type: facebook`

Here's the script producing the error

```
const horizon = Horizon({
  host: 'localhost:8181',
  authType: 'token'
})

if (!horizon.hasAuthToken()) { 
  horizon.authEndpoint('facebook').subscribe((endpoint) => { 
    window.location.pathname = endpoint
  })
} else {
  // We have a token already, do authenticated Horizon stuff here
}
```

And here's the result from `https://localhost:8181/horizon/auth_methods`
`{"facebook":"/horizon/facebook","github":"/horizon/github"}`

**Comment by deontologician** [comment]
What's are your auth settings in config.toml? (excluding secrets of course)

**Comment by deontologician** [comment]
Closing for now since there's not much to go on

**Comment by Grmiade** [comment]
I have the same error: `Unconfigured auth type: facebook`

My `config.tom` file:

```
token_secret = "*****"
# allow_anonymous = false
# allow_unauthenticated = false
auth_redirect = "/dashboard"

[auth.facebook]
id = "1034343653319829"
secret = "****"
```

and my code:

``` javascript
horizon({
  host: 'localhost:8181',
  authType: 'token'
})
```

``` javascript
if (!this.context.horizon.hasAuthToken()) {
  this.context.horizon.authEndpoint('facebook').subscribe(endpoint => {
    window.location.pathname = endpoint; // eslint-disable-line no-undef
  });
}
```

My @horizon/client version is 1.1.3
I checked `http://localhost:8181/horizon/auth_methods`, it's correct.

**Comment by Grmiade** [comment]
I found the problem.
Horizon calls `http://localhost:3000/horizon/auth_methods` instead of `http://localhost:8181/horizon/auth_methods`. (Yes, I don't serve my project with `hz serve`) 
The `host` option should be used to determine the url, no?

**Comment by danielmewes** [comment]
@Grmiade I think you ran into https://github.com/rethinkdb/horizon/issues/422 there. That issue will be fixed in the upcoming 2.0.0 release this week.

**Comment by luandro** [comment]
So until 2.0.0 there's no way to have a front-end app separated from the horizon server? Shouldn't any of these issues be kept open until this is fixed?

**Comment by danielmewes** [comment]
@luandro We usually close issues once the implementation is done and they have been merged into one of the main branches (in this case the `next` branch, which is going to be the basis for Horizon 2.0.0). This allows us to use issues to keep track of what work still needs to be done before a release.

**Comment by luandro** [comment]
Oh ok, got it.
